atomisticparsers/gromacs/parser.py

```diff
@@ -810,19 +810,66 @@ def get_gromacs_file(self, ext):
 
         return os.path.join(self._maindir, files[counts.index(min(counts))])
 
+    def get_all_gromacs_files(self, ext):
+        files = [d for d in self._gromacs_files if d.endswith(ext)]
+
+        if len(files) == 0:
+            return ''
+
+        all_files = {}
+        contain_basename, no_basename_match, counts = [], [], []
+        for f in files:
+            # We assume that the matching trajectory file has (or starts with)
+            # the same basename as the log file, e.g., out.log would correspond to
+            # out.tpr and out.trr and out.edr (or out_some.tpr ...).
+            # These files are therefore given first (second... ) priority in the dictionary.
```

Comment:
can you clean these comments a bit, they are a bit verbose and more confusing than anything. I think you just need to say something like "sort files with .ext into 3 priority categories" at the top, then briefly describe each case directly above the if or elif statement